.github/workflows/svc-bgs-api-integration-test.yml

```diff
@@ -0,0 +1,174 @@
+          body: '{"durable":true, "auto_delete":true}'
+          accept: 201, 204
+          retries: 3
+          log-response: true
+
+      # Temporary step added to avoid race condition. Currently, there is no health check in 'svc-bgs-api'; therefore,
+      # at this step, it is unknown if svc-bgs-api has yet to create the binding from 'bgs-api' exchange to 'add-note'
+      # queue.
+      - name: 'Create binding for add-note queue to bgs-api exchange'
+        uses: indiesdev/curl@v1.1
+        with:
+          url: 'http://localhost:15672/api/bindings/%2f/e/bgs-api/q/add-note'
+          method: 'POST'
+          basic-auth-token: '${{env.RABBITMQ_BASIC_AUTH}}'
+          body: '{"routing_key":"add-note","arguments":{}}'
+          accept: 201, 204
+          retries: 3
+          log-response: true
+
+      - name: 'Create add-note-response queue'
+        uses: indiesdev/curl@v1.1
+        with:
+          url: 'http://localhost:15672/api/queues/%2f/add-note-response'
+          method: 'PUT'
+          basic-auth-token: '${{env.RABBITMQ_BASIC_AUTH}}'
+          body: '{"durable":true, "auto_delete":true}'
+          accept: 201, 204
+          retries: 3
+          log-response: true
+
+      - name: 'Create binding for add-note-response queue to bgs-api exchange'
+        uses: indiesdev/curl@v1.1
+        with:
+          url: 'http://localhost:15672/api/bindings/%2f/e/bgs-api/q/add-note-response'
+          method: 'POST'
+          basic-auth-token: '${{env.RABBITMQ_BASIC_AUTH}}'
+          body: '{"routing_key":"add-note-response","arguments":{}}'
+          accept: 201, 204
+          retries: 3
+          log-response: true
+
+      - name: 'Send message to RabbitMQ on bgs-api exchange routed to add-note queue'
+        uses: indiesdev/curl@v1.1
+        with:
+          url: 'http://localhost:15672/api/exchanges/%2F/bgs-api/publish'
+          method: 'POST'
+          basic-auth-token: '${{env.RABBITMQ_BASIC_AUTH}}'
+          body: '{"properties": {"delivery_mode": 1,"reply_to": "add-note-response","correlation_id": "${{env.CORRELATION_ID}}"},"routing_key": "add-note","payload": "{\\\"veteranNote\\\":\\\"test\\\",\\\"veteranParticipantId\\\":111}","payload_encoding": "string"}'
+          accept: 200
+          retries: 3
+          log-response: true
+
+      - name: 'Sleep to give svc-bgs-api time to process and put response in add-note-response queue'
+        run: sleep 5
+        shell: bash
+
+      - name: 'Get message from RabbitMQ add-note-response queue'
+        id: 'addNoteResponse'
+        uses: indiesdev/curl@v1.1
+        with:
+          url: 'http://localhost:15672/api/queues/%2f/add-note-response/get'
+          method: 'POST'
+          basic-auth-token: '${{env.RABBITMQ_BASIC_AUTH}}'
+          headers: '{"accept":"application/json","content-type":"application/json"}'
+          body: '{"count":1,"ackmode":"ack_requeue_true","encoding":"auto"}'
+          accept: 200
+          retries: 3
```

Comment:
Unfortunately, the get message still returns 200 if there is no messages in the queue, just with a response with an empty `data` object.

Reply:
I see. Hmmm. That means if the GH runner is slow such that there's no message in the queue, this step will still pass because it's not checking the response body.

Reply:
Non-blocking comment: this test can be improved by checking the response body.